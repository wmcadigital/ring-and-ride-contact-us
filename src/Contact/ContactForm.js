import { useState } from "react";
import FormContentWrapper from "../common/FormContentWrapper";
import BreadCrumb from "../common/BreadCrumb";
import FormWizard from "../common/FormWizard";
import EnquiryTypeQuery from "./EnquiryTypeQuery";
import EnquiryDetail from "./EnquiryDetail";
import CheckAnswers from "./CheckAnswers";
import NameEntry from "./NameEntry";
import RegistrationNoEntry from "./RegistrationNoEntry";
import ContactPreferenceEntry from "./ContactPreferenceEntry";
import Header from "../common/Header";
import {
  validateContactPreferences,
  validateCheckAnswers,
} from "../common/validation";

const sleep = (ms) => new Promise((resolve) => setTimeout(resolve, ms));

const onSubmit = async (values) => {
  await sleep(300);
  window.alert(JSON.stringify(values, 0, 2));
};

const ContactForm = () => {
  const [goToPage, setGoToPage] = useState(null);

  return (
    <>
      <Header heading="Contact us" />
      <FormContentWrapper>
        <BreadCrumb currentPageName="Contact us" />
        <FormWizard
          onSubmit={onSubmit}
          goToPage={goToPage}
          setGoToPage={setGoToPage}
        >
          <EnquiryTypeQuery />
          <EnquiryDetail />
          <NameEntry />
          <RegistrationNoEntry />
          <ContactPreferenceEntry validate={validateContactPreferences} />
          <CheckAnswers
            setGoToPage={setGoToPage}
            validate={validateCheckAnswers}
          />
        </FormWizard>
      </FormContentWrapper>
    </>
  );
};

export default ContactForm;
