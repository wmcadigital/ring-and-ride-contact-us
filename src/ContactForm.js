const ContactForm = () => {
  return <div>Contact Form Placeholder</div>;
};

export default ContactForm;
