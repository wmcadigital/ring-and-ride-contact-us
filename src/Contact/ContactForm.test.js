import { create } from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

import ContactForm from "./ContactForm";

describe("ContactForm", () => {
  it("renders Contact Form as expected", () => {
    const renderer = create(
      <BrowserRouter>
        <ContactForm />
      </BrowserRouter>
    );

    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
