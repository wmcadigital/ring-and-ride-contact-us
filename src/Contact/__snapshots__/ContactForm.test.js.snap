// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm renders Contact Form as expected 1`] = `
<React.Fragment>
  <Header
    heading="Contact us"
  />
  <FormContentWrapper>
    <BreadCrumb
      currentPageName="Contact us"
    />
    <FormWizard
      goToPage={null}
      initialValues={Object {}}
      onSubmit={[Function]}
      setExternalPage={[Function]}
      setGoToPage={[Function]}
    >
      <EnquiryTypeQuery />
      <EnquiryDetail />
      <NameEntry />
      <RegistrationNoEntry />
      <ContactPreferenceEntry
        validate={[Function]}
      />
      <CheckAnswers
        setGoToPage={[Function]}
        validate={[Function]}
      />
    </FormWizard>
  </FormContentWrapper>
</React.Fragment>
`;
